### PR TITLE
DataFlash: zero-pad the number in the log filename

### LINKED ISSUE #2212: wishlist: data logs should be saved in filenames with leading zeros
Currently I get 1.BIN 9.BIN 10.BIN. 100.BIN...
This doesn't sort well in ls.
Can that be changed to 0000001.BIN 0000010.BIN etc ?

### PR DESCRIPTION
**Pull request by peterbarker**
This helps with collating when reviewing log files

Closes #2212

### CODE REVIEW

**Comment by WickedShell** [comment]
Haven't looked at implementation, but I'm excited for it, simple change that makes it a lot easier to find logs on some computers :)

**Comment by tridge** [comment]
have you tested this with the various GCS implementations?

File: libraries/DataFlash/DataFlash_File.cpp [hunk 1]
```diff
@@ -429,6 +430,39 @@ char *DataFlash_File::_log_file_name(const uint16_t log_num) const
...
 
 /*
+  construct a log file name given a log number.
+  The number in the log filename will be zero-padded.
+  Note: Caller must free.
+ */
+char *DataFlash_File::_log_file_name_long(const uint16_t log_num) const
+{
+    char *buf = nullptr;
+    if (asprintf(&buf, "%s/%08u.BIN", _log_directory, (unsigned)log_num) == 0) {
```
Comment (muramura) [inline libraries/DataFlash/DataFlash_File.cpp]: It returns -1 if the asprintf method detects an error. It is not 0.

**Comment by peterbarker** [comment]
@tridge shouldn't change what goes over mavlink as we kind-of virtualize the numbers there.

File: libraries/DataFlash/DataFlash_File.cpp (hunk 1 above)
Comment (peterbarker) [inline libraries/DataFlash/DataFlash_File.cpp]: @muramura Wow, well spotted.

In my defense, I was just being *consistently* wrong with what was already in the file :-)

I've pushed up a patch beneath this one to fix the pre-existing problems, and corrected this one to not include the problem.

Thanks!

**Pull request merged**